### DISCUSSION TITLE
Espressif component manager and registry

### DISCUSSION THREAD

**Discussion by beriberikix** [UncleRus/esp-idf-lib#465] [Ideas]
Are there plans to publish to the new [Component Registry](https://components.espressif.com/) from Espressif, as documented [here](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/tools/idf-component-manager.html)?

**Comment by UncleRus** [comment]
The documentation you linked to is well known and contains only a description of the component manifest, but nothing about registration in the registry.

We have repeatedly contacted Espressif about publishing components in their registry, but:

1. Currently, Espressif does not allow third parties to register their components in the registry.
2. Espressif is unable to ensure the security of the registry nor the legal basis.

**Reply by igrr** [reply]
Hi @UncleRus, could you please explain the 2nd point in more detail? If there are any concerns about the security or the legal aspects of the registry I'd like to address them. Is that related to the mention of kik/leftpad in [this comment](https://github.com/UncleRus/esp-idf-lib/issues/260#issuecomment-977919838)?

(On the 1st point, I'm also checking if we can open the registry access for esp-idf-lib project ahead of the public availability.)

**Reply by trombik** [reply]
@igrr i guess @UncleRus has his concerns. here is mine.

yes, the kik and leftpad debacles are my concerns. even worse, platformio did worse in its so-called registry (i stopped using platformio for serious projects, it might be better now. but i don't know). anyone can upload their work, or even *a fork*, just because they want to use it in platformio, but the author doesn't want to get involved in the fuss. the registry might be *the largest* in the world, but it is very hard to confirm its authenticity. you, as a vendor, might want more people to join, and lower the hurdle. but, as a user, I don't want to use random libraries from unknown sources (i'm referring authors, rather than URLs, or source repositories). Arduino probably has similar issues, but I don't know because I don't use it.

you might have been doing already, but you should do risk assessment in the project. as espressif has been demonstrating how serious it is in security, i am sure it will do the same in the component registry. 

I've just got a trial account for my own project. I'll explore the component registry and give you my feedback.

**Comment by beriberikix** [comment]
I appreciate the quick response! I just learned about the registry this week from their online event, my apologies. They mentioned thy are planning on opening it up to third parties soon - maybe @igrr might know more?

**Reply by kumekay** [reply]
@beriberikix  @UncleRus Last week we opened the registry for 3rd party contributors in preview-mode. If you are interested feel free to fill the form   https://components.espressif.com/preview/

**Comment by trombik** [comment]
see also: https://github.com/espressif/idf-component-manager/issues/4

**Comment by lukilukeskywalker** [comment]
Hi! 
I am wondering if there has been any progress on this? Having this set of components in the esp idf component registry would facilitate the management of the components themselves.
Also I have been thinking that separating each of the components in it's own git repository, would facilitate the integration of the components in the users projects. you could join all the components with the actual structure with submodules, each of the folders pointing to it's own repository. 
This way, we wouldn't need to download the whole repository with components we might not need, and we would be able to integrate the components inside the project with idf.py add-dependency ssh:git@[...] as stated in [this doc](https://espressif-docs.readthedocs-hosted.com/_/downloads/idf-component-manager/en/latest/pdf/)  (page 7)

**Reply by lukilukeskywalker** [reply]
Oh, my bad, just saw you can specify the path to a component inside a github repository inside the idf_component.yml with 
`  UncleRus/i2cdev: `
    `git: https://github.com/UncleRus/esp-idf-lib.git`
    `path: components/i2cdev`

**Reply by trombik** [reply]
I'm not so active but have been following the development of idf-component. there have been many improvements but i do not consider they are stable. the migration to multiple repositories involves many tasks you, as a user, have never imagined.
